x/axelarcork/keeper/msg_server.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/peggyjv/sommelier/v6/x/axelarcork/types"
+)
+
+var _ types.MsgServer = Keeper{}
+
+// ScheduleCork implements types.MsgServer
+func (k Keeper) ScheduleCork(c context.Context, msg *types.MsgScheduleAxelarCorkRequest) (*types.MsgScheduleAxelarCorkResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	if !k.GetParamSet(ctx).Enabled {
+		return nil, types.ErrDisabled
+	}
+
+	config, err := k.GetChainConfigurationByNameAndID(ctx, msg.ChainName, msg.ChainId)
+	if err != nil {
+		return nil, err
+	}
+
+	if !k.HasCellarID(ctx, config.Id, common.HexToAddress(msg.Cork.TargetContractAddress)) {
+		return nil, types.ErrUnmanagedCellarAddress
+	}
+
+	if msg.BlockHeight <= uint64(ctx.BlockHeight()) {
+		return nil, types.ErrSchedulingInThePast
+	}
+
+	signer := msg.MustGetSigner()
+	validator := k.stakingKeeper.Validator(ctx, sdk.ValAddress(signer))
```

Comment:
Wait, axelarcork now depends on the gravity keeper? So the whole point of splitting them out because they technically referenced other modules no longer matters? This should 100% be part of cork v2 if that is the case

Reply:
I don't recall referencing other modules being the reasoning here. But axelarcork messages are submitted in a similar fashion to regular corks, because steward only has access to the orchestrator key. Having them as separate modules though allows e.g. divergent functionality and disparate param sets, I don't think we need to move anything.